core/src/main/java/org/apache/iceberg/stats/BaseContentStats.java

```diff
@@ -0,0 +1,205 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.stats;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.iceberg.data.GenericRecord;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+public class BaseContentStats implements ContentStats, Serializable {
+
+  private final List<FieldStats<?>> fieldStats;
+
+  /** Used by Avro reflection to instantiate this class when reading manifest files. */
+  public BaseContentStats(Types.StructType projection) {
+    this.fieldStats = Lists.newArrayListWithCapacity(projection.fields().size());
+    for (int i = 0; i < projection.fields().size(); i++) {
+      Types.NestedField field = projection.fields().get(i);
+      Preconditions.checkArgument(
+          field.type().isStructType(), "Field stats must be a struct type: %s", field.type());
+      Types.StructType structType = field.type().asStructType();
+      Type type = null;
+      if (null != structType.field("lower_bound")) {
+        type = structType.field("lower_bound").type();
+      } else if (null != structType.field("upper_bound")) {
+        type = structType.field("upper_bound").type();
+      }
+
+      fieldStats.add(
+          BaseFieldStats.builder()
+              .fieldId(StatsUtil.fieldIdForStatsField(field.fieldId()))
+              .type(type)
+              .build());
+    }
+  }
+
+  private BaseContentStats(List<FieldStats<?>> fieldStats) {
+    this.fieldStats = Lists.newArrayList(fieldStats);
+  }
+
+  @Override
+  public List<FieldStats<?>> fieldStats() {
+    return fieldStats;
+  }
+
+  @Override
+  public int size() {
+    return fieldStats.size();
+  }
+
+  @Override
+  public <T> T get(int pos, Class<T> javaClass) {
+    if (pos < 0 || pos > fieldStats().size() - 1) {
+      return null;
+    }
+
+    FieldStats<?> value = fieldStats.get(pos);
+    if (value == null || javaClass.isInstance(value)) {
+      return javaClass.cast(value);
+    }
+
+    throw new IllegalArgumentException(
+        String.format(
+            "Wrong class, expected %s but was %s for object: %s",
+            javaClass.getName(), value.getClass().getName(), value));
+  }
+
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  @Override
+  public <T> void set(int pos, T value) {
+    if (value instanceof GenericRecord) {
+      GenericRecord record = (GenericRecord) value;
```

Comment:
I assume we have a use case where the field stats was captured as a `GenericRecord`. wondering if we can move the code that convert from GenericRecord to BaseFieldStats in a util method. maybe a static method like `BaseFieldStats.from(GenericRecord)`?

Reply:
see my other [comment](https://github.com/apache/iceberg/pull/13933#discussion_r2413779883) on this. I don't think it helps by introducing a new util method, since we don't have any other use cases for this